### PR TITLE
Validate CompressionLevel value is defined in BrotliStream ctor

### LINKED ISSUE #46595: Revisit BrotliStream CompressionLevel values
Two independent-but-related issues:

1.  The ctor:
```C#
public BrotliStream(Stream stream, CompressionLevel compressionLevel)
```
passes the `compressionLevel` to the function:
```C#
        internal static int GetQualityFromCompressionLevel(CompressionLevel level) =>
            level switch
            {
                CompressionLevel.Optimal => Quality_Default,
                CompressionLevel.NoCompression => Quality_Min,
                CompressionLevel.Fastest => 1,
                CompressionLevel.SmallestSize => Quality_Max,
                _ => (int)level,
            };
```
with no argument validation.  If an arbitrary level is provided, that then gets passed through as-is to the underlying library, resulting in inconsistent and potentially unexpected behavior.  We should decide whether we're ok with this behavior or whether we should take a breaking change to correct it and be more like DeflateStream, which throws for an unknown value.

2. The quality values are set as:
```
        public const int Quality_Min = 0;
        public const int Quality_Default = 11;
        public const int Quality_Max = 11;
```
With the introduction of SmallestSize (https://github.com/dotnet/runtime/issues/1549#issue-547746629) and the rationale employed for it that Optimal should really be a good balance between speed and size, we should define it closer to 6 rather than 11.  The docs will also need to be updated.

### PR DESCRIPTION
**Pull request by i3arnon**
Fix #46595

### CODE REVIEW

**Comment by i3arnon** [comment]
> 2\. Ask a committer to mail the `.NET Breaking Change Notification` DL.

@stephentoub Can you please send the breaking change suggestion to the DL? dotnet/docs#25665

**Comment by stephentoub** [comment]
_if_ it's merged

**Comment by carlossanlop** [comment]
The CI results are already deleted. I want to make sure we don't have unit test failures. Closing and reopening to re-trigger the CI (same with the other related PR).

**Review by jeffhandley — CHANGES_REQUESTED** [review]
Please see the comment from @scalablecory here: https://github.com/dotnet/runtime/pull/57561#discussion_r723748527

File: src/libraries/System.IO.Compression.Brotli/src/System/IO/Compression/enc/BrotliStream.Compress.cs [hunk 1]
```diff
@@ -23,6 +23,17 @@ public sealed partial class BrotliStream : Stream
         /// <param name="leaveOpen"><see langword="true" /> to leave the stream open after disposing the <see cref="System.IO.Compression.BrotliStream" /> object; otherwise, <see langword="false" />.</param>
         public BrotliStream(Stream stream, CompressionLevel compressionLevel, bool leaveOpen) : this(stream, CompressionMode.Compress, leaveOpen)
         {
+            switch (compressionLevel)
```
[the remark this answers is not in the document]
Comment (carlossanlop) [reply src/libraries/System.IO.Compression.Brotli/src/System/IO/Compression/enc/BrotliStream.Compress.cs]: @i3arnon hope you don't mind - I pushed the proposed change to your branch and I added tests to validate it.

[the remark this answers is not in the document]
Comment (carlossanlop) [reply src/libraries/System.IO.Compression.Brotli/src/System/IO/Compression/enc/BrotliStream.Compress.cs]: @i3arnon hope you don't mind - I pushed the proposed change to your branch and I added tests to validate it.

**Review by carlossanlop — APPROVED** [review]
I'm signing off, although I added the last commits to address the suggestion and to add tests. @Jozkee @adamsitnik @jeffhandley @stephentoub would you mind giving us an additional sign-off if you agree with the change?

**Review by jeffhandley — APPROVED** [review]

**Comment by ericstj** [comment]
@carlossanlop were you planning to create a breaking change document for this?

**Comment by eerhardt** [comment]
We should also consider updating this blog: https://devblogs.microsoft.com/dotnet/introducing-support-for-brotli-compression/

It explicitly calls out this behavior as a feature:

> As mentioned earlier, Brotli supports 11 levels of compression quality aside from the standard Optimal, Fastest and NoCompression levels. You can pass in the level to the constructor by casting the desired level directly to CompressionLevel, like this:
```C#
BrotliStream brotli = new BrotliStream(baseStream,
                                       CompressionMode.Compress,
                                       leaveOpen,
                                       bufferSize,
                                       (CompressionLevel)5); // Use level 5
```

I stumbled on it from searching for "brotli compression kestrel".

**Comment by stephentoub** [comment]
I've removed it.

**Comment by Jozkee** [comment]
breaking change doc created https://github.com/dotnet/docs/issues/32620.

**Pull request merged**